pkg/operator/controller/canary/route_test.go

```diff
@@ -106,7 +110,10 @@ func TestCanaryRouteChanged(t *testing.T) {
 	service := desiredCanaryService(daemonsetRef)
 
 	for _, tc := range testCases {
-		original := desiredCanaryRoute(service)
+		original, err := desiredCanaryRoute(service)
+		if err != nil {
```

Comment:
Consider adding a test case that produces an error, and checking tc.expect to make sure the error is expected.

Reply:
I thought about adding these, but didn't in the interest of time. Maybe that would be a good thing to add in a future cleanup? I could add these test cases to my current cleanup PR, https://github.com/openshift/cluster-ingress-operator/pull/520.

Reply:
It should be quick to add a new function to exercise the validation in desiredCanaryRoute, and check that an error was produced for invalid input.  It doesn't have to be a part of this function.

Reply:
See https://github.com/openshift/cluster-ingress-operator/pull/520/commits/d6462c265642771af4be6c992af662c7db342a0c